contracts/packages/Spawner.sol

```diff
@@ -93,24 +93,18 @@ contract Spawner {
             type(Spawn).creationCode,
             abi.encode(logicContract, initializationCalldata)
         );
-
         // get target address using the constructed initialization code.
         bytes32 initCodeHash = keccak256(initCode);
 
-        target = address( // derive the target deployment address.
-            uint160( // downcast to match the address type.
-                uint256( // cast to uint to truncate upper digits.
-                    keccak256( // compute CREATE2 hash using 4 inputs.
-                        abi.encodePacked( // pack all inputs to the hash together.
-                            bytes1(0xff), // pass in the control character.
-                            address(this), // pass in the address of this contract.
-                            bytes32(SALT), // pass in the salt from above.
-                            initCodeHash // pass in hash of contract creation code.
-                        )
-                    )
-                )
+        bytes32 _data = keccak256( // compute CREATE2 hash using 4 inputs.
+            abi.encodePacked(
+                bytes1(0xff), // pass in the control character.
+                address(this), // pass in the address of this contract.
+                SALT, // pass in the salt from above.
+                initCodeHash // pass in hash of contract creation code.
             )
         );
+        target = address(uint256(_data));
```

Comment:
In simplifying _computeAddress, there is a missing uint160 cast, can u explain why u removed it ?

Reply:
OpenZeppelin has a [CREATE2 library](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Create2.sol), they use this cast and there's a small discussion 
https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2244 on this topic. The `address()` cast will take care of the uint256 => uint160.